### PR TITLE
SDIT-2250 Revised case notes from-nomis sync logic

### PR DESCRIPTION
**Pull request by steverendell**
Taking previous merges into account

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesByPrisonerMigrationService.kt [hunk 1]
```diff
@@ -84,10 +84,13 @@ class CaseNotesByPrisonerMigrationService(
...
           }
           if (originals.isEmpty()) {
-            log.warn("No duplicates found for offenderNo: $offenderNo, merged case note ${mergeCopiedCaseNote.caseNoteId}")
+            log.warn("No original found for offenderNo: $offenderNo, merged case note ${mergeCopiedCaseNote.caseNoteId}")
           } else {
             copiedNomisIdSet.add(mergeCopiedCaseNote.caseNoteId)
           }
+          if (originals.size > 1) {
+            log.warn("More than 1 originals found for offenderNo: $offenderNo, merged case note ${mergeCopiedCaseNote.caseNoteId}")
+          }
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesByPrisonerMigrationService.kt:93]: migration code tweak - I dont think this should happen so would like to know if it does

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesNomisApiService.kt [hunk 2]
```diff
@@ -21,4 +23,12 @@ class CaseNotesNomisApiService(@Qualifier("nomisApiWebClient") private val webCl
...
       .retrieve()
       .awaitBody()
+
+  suspend fun updateCaseNote(caseNoteId: Long, nomisCaseNote: UpdateCaseNoteRequest) {
+    webClient.put()
+      .uri("/casenotes/{caseNoteId}", caseNoteId)
+      .bodyValue(nomisCaseNote)
+      .retrieve()
+      .awaitBodilessEntity()
+  }
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesNomisApiService.kt:33]: The casenotes sync now does something unusual - when a Nomis case note changes, it potentially updates a related case note in Nomis

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesMappingApiMockServer.kt [hunk 3]
```diff
@@ -114,10 +114,6 @@ class CaseNotesMappingApiMockServer(private val objectMapper: ObjectMapper) {
     )
   }
 
-  fun stubPostBatchMappingsFailureFollowedBySuccess() {
-    mappingApi.stubMappingCreateFailureFollowedBySuccess(url = "/mapping/casenotes/batch")
-  }
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesMappingApiMockServer.kt:119]: not used

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesNomisApiService.kt (hunk 2 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesNomisApiService.kt:33]: Will that then raise an xtag event which will then be ignored?

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**